packages/flutter_tools/lib/src/asset.dart

```diff
@@ -113,8 +113,9 @@ class AssetBundle {
         manifestDescriptor['uses-material-design'];
 
     for (_Asset asset in assetVariants.keys) {
-      assert(asset.assetFileExists);
-      entries[asset.assetEntry] = new DevFSFileContent(asset.assetFile);
+      assert(asset.assetFileExists || assetVariants.isNotEmpty);
```

Comment:
If this is just an invariant that you're checking, leave a comment pointing to where we do the input validation.